File: DSA/Yashvardhan-Guliyani/Day12_q2.java
public class SumDigitsUntilSingleDigit {
    public static int sumDigits(int n) {
        if (n < 10) {
            return n;
        } else {
            return n % 10 + sumDigits(n / 10);
        }
    }

    public static int sumDigitsUntilSingleDigit(int n) {
        int result = sumDigits(n);
        while (result >= 10) {
            result = sumDigits(result);
        }
        return result;
    }

    public static void main(String[] args) {
        int number = 9875;
        System.out.println("The single digit result for " + number + " is: " + sumDigitsUntilSingleDigit(number));
    }
}

File: DSA/saumya/DAY12_Q2.py
def single_digit_sum(number):
  if number < 10:
    return number
  else:
    sum_of_digits = 0
    while number > 0:
      last_digit = number % 10
      sum_of_digits += last_digit
      number //= 10
    return single_digit_sum(sum_of_digits)  # Recursive to handle remaining digits

# Example usage
number = int(input("Enter a number: "))
result = single_digit_sum(number)
print(f"The single digit sum of {number} is: {result}")
